hazelcast/src/main/java/com/hazelcast/map/impl/query/MapLocalParallerQueryRunner.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.query;
+
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.map.impl.MapServiceContext;
+import com.hazelcast.query.PagingPredicate;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.impl.QueryableEntry;
+import com.hazelcast.query.impl.predicates.QueryOptimizer;
+import com.hazelcast.util.executor.ManagedExecutorService;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static com.hazelcast.map.impl.query.MapQueryEngineUtils.waitForResult;
+import static com.hazelcast.query.PagingPredicateAccessor.getNearestAnchorEntry;
+import static com.hazelcast.util.SortingUtil.getSortedSubList;
+
+/**
+ * Specialization of the {@link MapLocalQueryRunner} - the only difference is that
+ * the query evaluation per partition is run in a PARALLEL fashion.
+ *
+ * Runs query operations in the calling thread (thus blucking it)
+ * Query evaluation per partition is run in a PARALLEL fashion.
+ * <p>
+ * Used by query operations only: QueryOperation & QueryPartitionOperation
+ * Should not be used by proxies or any other query related objects.
+ */
+public class MapLocalParallerQueryRunner extends MapLocalQueryRunner {
+
+    protected static final int QUERY_EXECUTION_TIMEOUT_MINUTES = 5;
+
+    private final ManagedExecutorService executor;
+    private final int timeoutInMinutes;
+
+    public MapLocalParallerQueryRunner(MapServiceContext mapServiceContext, QueryOptimizer optimizer,
+                                       ManagedExecutorService executor, int timeoutInMinutes) {
+        super(mapServiceContext, optimizer);
+        this.executor = executor;
+        this.timeoutInMinutes = timeoutInMinutes;
+    }
+
+    public MapLocalParallerQueryRunner(MapServiceContext mapServiceContext, QueryOptimizer optimizer,
+                                       ManagedExecutorService executor) {
+        this(mapServiceContext, optimizer, executor, QUERY_EXECUTION_TIMEOUT_MINUTES);
+    }
+
+    @Override
+    protected Collection<QueryableEntry> runPartitionScanQuery(
+            String mapName, Predicate predicate, Collection<Integer> partitions) {
+        try {
+            if (predicate instanceof PagingPredicate) {
+                return runParallelFullPartitionScanQueryWithPaging(mapName, (PagingPredicate) predicate, partitions);
+            } else {
+                return runParallelFullPartitionScanQueryWithoutPaging(mapName, predicate, partitions);
```

Comment:
Do we still need the 'runParallel' prefix? Since this is already the parallel runner.